sdk/core/src/services/github/service.rs

```diff
@@ -0,0 +1,141 @@
+
+use crate::services::github::error::Error;
+use crate::services::github::client::{
+    Client as GitHubClient,
+    Repo,
+};
+
+/// Definition of the GitHubProvider
+#[derive(Debug)]
+pub struct GitHubService {
+    org: String,
+    client: GitHubClient,
+}
+
+/// Impl for GitHubSbomProvider
+///
+impl GitHubService {
+    /// new method sets the organization for the struct
+    ///
+    pub fn new(org: String, pat: String) -> Self {
+        let client = GitHubClient::new(pat);
+        GitHubService {
+            org,
+            client,
+        }
+    }
+
+    /// Use the GitHub Client to get all of the repositories
+    /// from the GitHub Organization
+    pub(crate) async fn get_repos(&self) -> Result<Vec<Repo>, Error> {
+
+        let mut pages = match self.client.get_pages(&self.org).await {
+            Ok(pages) => pages,
+            Err(err) => return Err(
+                Error::GitHubErrorResponse(
+                    format!("Unable to get pages of repos from GitHub: {:#?}", err)
+                )
+            )
+        };
+
+        let mut repo_vec: Vec<Repo> = Vec::new();
+        for (page, per_page) in pages.iter_mut().enumerate() {
+
+            let mut gh_org_rsp = self.client.get_page_of_repos(&self.org, page + 1, per_page).await?;
+
+            for (repo_num, repo) in gh_org_rsp.iter_mut().enumerate() {
+
+                println!("Repo number: {}, ", repo_num);
+                let result = self.client.get_last_commit(repo).await;
+                let repo_name = repo.full_name.clone().unwrap();
+
+                match result {
+                    Ok(option) => match option {
+                        Some(last_hash) => repo.add_last_hash(last_hash),
+                        None => println!("No last commit has found for Repo: {}", &repo_name)
```

Comment:
This should continue or return an Error

Reply:
Does this just mean it's the first time we saw the repo?

Reply:
No, it shouldn't necessarily be the first time we saw it.  We need to get that last hash every time we look at the repo so we can determine if we should skip it or not.